src/core/server/csp/csp_directives.test.ts

```diff
@@ -0,0 +1,134 @@
+          "style-src foo bar",
+        ]
+      `);
+    });
+
+    it('properly updates the header', () => {
+      const directives = new CspDirectives();
+      directives.addDirectiveValue('style-src', 'foo');
+
+      expect(directives.getCspHeader()).toMatchInlineSnapshot(`"style-src foo"`);
+
+      directives.addDirectiveValue('style-src', 'bar');
+
+      expect(directives.getCspHeader()).toMatchInlineSnapshot(`"style-src foo bar"`);
+    });
+
+    it('handles distinct directives', () => {
+      const directives = new CspDirectives();
+      directives.addDirectiveValue('style-src', 'foo');
+      directives.addDirectiveValue('style-src', 'bar');
+      directives.addDirectiveValue('worker-src', 'self');
+
+      expect(directives.getCspHeader()).toMatchInlineSnapshot(
+        `"style-src foo bar; worker-src self"`
+      );
+      expect(directives.getRules()).toMatchInlineSnapshot(`
+        Array [
+          "style-src foo bar",
+          "worker-src self",
+        ]
+      `);
+    });
+
+    it('removes duplicates', () => {
+      const directives = new CspDirectives();
+      directives.addDirectiveValue('style-src', 'foo');
+      directives.addDirectiveValue('style-src', 'foo');
+      directives.addDirectiveValue('style-src', 'bar');
+
+      expect(directives.getCspHeader()).toMatchInlineSnapshot(`"style-src foo bar"`);
+      expect(directives.getRules()).toMatchInlineSnapshot(`
+        Array [
+          "style-src foo bar",
+        ]
+      `);
+    });
+  });
+
+  describe('#fromConfig', () => {
+    it('returns the correct rules for the default config', () => {
+      const config = cspConfig.schema.validate({});
+      const directives = CspDirectives.fromConfig(config);
+      expect(directives.getRules()).toMatchInlineSnapshot(`
+        Array [
+          "script-src 'unsafe-eval' 'self'",
+          "worker-src blob: 'self'",
+          "style-src 'unsafe-inline' 'self'",
+        ]
+      `);
+    });
+
+    it('returns the correct header for the default config', () => {
+      const config = cspConfig.schema.validate({});
+      const directives = CspDirectives.fromConfig(config);
+      expect(directives.getCspHeader()).toMatchInlineSnapshot(
+        `"script-src 'unsafe-eval' 'self'; worker-src blob: 'self'; style-src 'unsafe-inline' 'self'"`
+      );
+    });
+
+    it('handles config with rules', () => {
+      const config = cspConfig.schema.validate({
+        rules: [`script-src 'self' http://foo.com`, `worker-src 'self'`],
+      });
+      const directives = CspDirectives.fromConfig(config);
+
+      expect(directives.getRules()).toMatchInlineSnapshot(`
+        Array [
+          "script-src 'self' http://foo.com",
+          "worker-src 'self'",
+        ]
+      `);
+      expect(directives.getCspHeader()).toMatchInlineSnapshot(
+        `"script-src 'self' http://foo.com; worker-src 'self'"`
+      );
+    });
+
+    it('adds default value for config with directives', () => {
```

Comment:
Atm it doesn't without a TS violation because I hardcoded the subset of directives we're using:

`export type CspDirectiveName = 'script-src' | 'worker-src' | 'style-src' | 'frame-ancestors';`

Do you think I should use an exhaustive list instead?

Reply:
I probably should have commented on the comment above for better clarity. I don't think we necessarily need to have support for this in code / TS, but we need a test to be sure that we don't break cases where the deprecated `csp.rules` config has a value like:

```yaml
csp.rules:
  - default-src http://myfoo.com
```

I don't believe we'll create an TS violations for creating a test with a rules value like that and asserting against the results of `getCspHeader()`